Editor/Scripts/CodeGenerators/GenerateRGActionClasses.cs

```diff
@@ -0,0 +1,257 @@
+                                SyntaxFactory.Argument(methodExpression)
+                            )
+                        )
+                    );
+
+                return SyntaxFactory.Argument(delegateCreationExpression);
+            }
+            else
+            {
+                // When there are parameters, create the Action delegate type with type arguments
+                var actionType = SyntaxFactory.GenericName(
+                        SyntaxFactory.Identifier("Action")
+                    )
+                    .WithTypeArgumentList(
+                        SyntaxFactory.TypeArgumentList(
+                            SyntaxFactory.SeparatedList(
+                                action.Parameters.Select(param => SyntaxFactory.ParseTypeName(param.Type))
+                            )
+                        )
+                    );
+
+                // Create the delegate creation expression
+                var delegateCreationExpression = SyntaxFactory.ObjectCreationExpression(actionType)
+                    .WithArgumentList(
+                        SyntaxFactory.ArgumentList(
+                            SyntaxFactory.SingletonSeparatedList(
+                                SyntaxFactory.Argument(methodExpression)
+                            )
+                        )
+                    );
+
+                return SyntaxFactory.Argument(delegateCreationExpression);
+            }
+        }
+
+        private static MemberDeclarationSyntax GenerateStartActionMethod(RGActionInfo action)
+        {
+            var methodParameters = new List<ParameterSyntax>
+            {
+                SyntaxFactory.Parameter(SyntaxFactory.Identifier("input"))
+                             .WithType(SyntaxFactory.ParseTypeName("Dictionary<string, object>"))
+            };
+
+            var parameterParsingStatements = new List<StatementSyntax>();
+            var methodInvocationArguments = new List<string>();
+
+            foreach (var parameter in action.Parameters)
+            {
+                string paramName = parameter.Name;
+                string paramType = parameter.Type;
+
+                methodInvocationArguments.Add(paramName);
+
+                parameterParsingStatements.Add(SyntaxFactory.ParseStatement($"string {paramName}Input = input[\"{paramName}\"] as string;"));
+                parameterParsingStatements.Add(SyntaxFactory.ParseStatement($"{paramType} {paramName} = default;"));
+
+                string tryParseStatement;
+
+                if (paramType.ToLower() == "string" || paramType.ToLower() == "system.string")
+                {
+                    tryParseStatement = $"{paramName} = {paramName}Input;";
+                }
+                else if (RGUtils.IsCSharpPrimitive(paramType))
+                {
+                    tryParseStatement = $"{paramType}.TryParse({paramName}Input, out {paramName});";
+                }
```

Comment:
Instead of treating a string separately, can't we have the first pull from the `input[paramName]` be and `object` type, and then do a cast? Rather than a parse? Not a big deal, just curious if you tried that too.